src/libraries/System.Private.Xml.Linq/tests/misc/LoadSaveAsyncTests.cs

```diff
@@ -202,5 +202,68 @@ public static IEnumerable<object[]> RoundtripOptions_MemberData
             }
         }
 
+        public static IEnumerable<object[]> IsAsync_SaveOptions_Data
+        {
+            get
+            {
+                foreach (bool isAsync in new[] { true, false })
+                    foreach (SaveOptions saveOptions in Enum.GetValues(typeof(SaveOptions)))
+                        yield return new object[] { isAsync, saveOptions };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IsAsync_SaveOptions_Data))]
+        public async Task SaveAsync_CallsAsyncOnly_SaveSync_CallsSyncOnly(bool isAsync, SaveOptions saveOptions)
+        {
+            XDocument document = XDocument.Parse("<root>Test document async save</root>");
+            var element = new XElement("Test");
+            using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
+            {
+                if (isAsync)
+                {
+                    await document.SaveAsync(stream, saveOptions, CancellationToken.None);
+                    await element.SaveAsync(stream, saveOptions, CancellationToken.None);
+                }
+                else
+                {
+                    document.Save(stream);
+                    element.Save(stream);
+                }
+            }
+        }
+    }
+
+    public class ForceSyncAsyncStream : MemoryStream
+    {
+        private bool _isAsync;
+
+        public ForceSyncAsyncStream(bool async)
+        {
+            _isAsync = async;
+        }
+        public override void Flush()
+        {
+            Assert.False(_isAsync, "Sync operation not allowed");
+            base.Flush();
+        }
+
+        public override Task FlushAsync(CancellationToken cancellationToken)
+        {
+            Assert.True(_isAsync, "Async operation not allowed");
```

Comment:
```suggestion
            Assert.True(_isAsync, "Sync operation not allowed");
```

Reply:
Also: Should you call Sync Flush here?

Reply:
This will result in a failure log of

```
       Sync operation not allowed
       Expected: True
       Actual:   False
```

Which seems like the right change... but in case it isn't, I thought I'd give the full example.

Reply:
> Also: Should you call Sync Flush here or just forward it to base.FlushAsync?

No, `base.FlushAsync` calling sync `Flush`

Reply:
For suggestion: It implies if `FlushAsync` is called for Sync stream report "Async operation not allowed"

Reply:
@buyaa-n what I meant is that your message says exactly opposite to what you're asserting, i.e. here `Assert.True(_isAsync, ...` you're asserting that only Async is allowed but message says that it's not

Reply:
Unless you use different terminology, when you say "sync operation" I'm thinking "XML class operating in sync mode"

Reply:
Yes it is confusing, i did change that message myself several times it still not sound perfect, how about changing it `Async call not allowed for sync stream`?